packages/lib-utils/src/types/api-discovery.ts

```diff
@@ -1,11 +1,12 @@
-import type { Store, AnyAction } from 'redux';
-import type { ActionType as Action } from 'typesafe-actions';
 import type { K8sVerb, K8sModelCommon } from './k8s';
 
-export type InitAPIDiscovery = (
-  store: Store<unknown, Action<AnyAction>>,
-  preferenceList?: string[],
-) => void;
+export type ApiActions = {
+  setResourcesInFlight: (isInFlight: boolean) => void;
+  setBatchesInFlight: (isInFlight: boolean) => void;
+  receivedResources: (resource: DiscoveryResources) => void;
+};
+
+export type InitAPIDiscovery = (apiActions: ApiActions, preferenceList?: string[]) => void;
```

Comment:
@karelhala do we really need to modify the signature of `InitAPIDiscovery`? I was thinking if an app chooses to use our baked-in API discovery they would just provide the `store`. If an app chooses not to use the built in API discovery we will have action creators exposed to pipe data into the SDK redux.

Reply:
Well we are calling the `apiDiscovery` which defaults to `initAPIDiscovery`. We can expose action creator and app would pass in the `store` to it and get these functions, which could probably be more straight forward approach. I was just thing of how to provide the proper store. But when we call the `apiDiscovery` we were passing the store so that could be used for that. Let me change it slightly.